hpobench/benchmarks/ml_mmfb/tabular_benchmark.py

```diff
@@ -0,0 +1,230 @@
+        #     mask *= df[param].values == row_dict[param]
+        # idx = np.where(mask)
+        # if len(idx) != 1:
+        #     return None
+        # idx = idx[0][0]
+        # result = df.iloc[idx]["result"]
+        # return result
+
+    @staticmethod
+    def _build_query(row_dict: Dict) -> str:
+        query = ''
+        for i, (param_name, param_value) in enumerate(row_dict.items()):
+            if i != 0:
+                query += ' & '
+            query += f'{param_name} == {param_value}'
+        return query
+
+    def _objective(
+            self,
+            config: Dict,
+            fidelity: Dict,
+            seed: Union[int, None] = None,
+            metric: Union[str, None] = "acc",
+            evaluation: Union[str, None] = ""
+    ) -> Dict:
+
+        metric_str = ', '.join(list(metrics.keys()))
+        assert metric in list(metrics.keys()), f"metric not found among: {metric_str}"
+        score_key = f"{evaluation}_scores"
+        cost_key = f"{evaluation}_scores"
+
+        key_path = dict()
+        # TODO: Dicts are unordered. This does not have to have an effect.
+        for name in np.sort(self.configuration_space.get_hyperparameter_names()):
+            key_path[str(name)] = config[str(name)]
+        for name in np.sort(self.fidelity_space.get_hyperparameter_names()):
+            key_path[str(name)] = fidelity[str(name)]
+
+        if seed is not None:
+            assert seed in self._seeds_used()
+            seeds = [seed]
+        else:
+            seeds = self._seeds_used()
+
+        loss = []
+        costs = 0.0
+        info = dict()
+        for seed in seeds:
+            key_path["seed"] = seed
+            res = self._search_dataframe(key_path, self.table)
+            loss.append(1 - res["info"][score_key][metric])
+            costs += res["info"]["model_cost"] + res["info"][cost_key][metric]
+            info[seed] = res["info"]
+            key_path.pop("seed")
+        loss = np.mean(loss)
+        result = dict(function_value=loss, cost=costs, info=info)
+        return result
+
+
+class TabularBenchmark(BaseTabularBenchmark):
+    def __init__(self, model: str, task_id: int, data_dir: Union[Path, str, None] = None,
+                 rng: Union[int, np.random.RandomState, None] = None, **kwargs):
+        super(TabularBenchmark, self).__init__(model, task_id, data_dir, rng, **kwargs)
+
+    # pylint: disable=arguments-differ
+    def get_configuration_space(self, seed: Union[int, None] = None) -> CS.ConfigurationSpace:
+        cs = json_cs.read(self.config_spaces['x_discrete'])
+        cs = self._preprocess_configspace(cs)
+        cs.seed(seed)
+        return cs
+
+    # pylint: disable=arguments-differ
+    def get_fidelity_space(self, seed: Union[int, None] = None) -> CS.ConfigurationSpace:
+        cs = json_cs.read(self.config_spaces['z_discrete'])
+        cs.seed(seed=seed)
+        return cs
+
+
+class OriginalTabularBenchmark(BaseTabularBenchmark):
```

Comment:
I don't think we need this class definition. I'd rather have the utilities of this class as a function along the lines of `get_original_configuration/fidelity_space` inside `TabularBenchmark`. 
Given that the `TabularBenchmark` is designed to query the tabular data using the discretized configspace, it doesn't make sense to instantiate an entire object of a different class with the `original` space since the `objective_function` call won't work with this `original` non-discretized space. 
Exposing `get_original_configuration/fidelity_space` is just for the user to see the original space that was used to derive the discrete space returned by the current definition inside `TabularBenchmark`. 

As a result, I think we can do away with this class, and combine the BaseClass and the TabularBenchmark into just one class.

Let me know if you still need this design for the sake of containerization and the need to inherit `AbstractBenchmark`.

Reply:
Ah, get it now! Thanks for the clarification. 
I have removed the second class. (original) and moved the original config space/fidelity space to the meta-information as a workaround. Maybe, we could open an issue and expose a new function in the container interface that enables us to query the function later.